atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractTransactionManager.java

```diff
@@ -28,8 +29,11 @@
 public abstract class AbstractTransactionManager implements TransactionManager {
     public static final Logger log = LoggerFactory.getLogger(AbstractTransactionManager.class);
 
+    protected boolean isClosed;
+
     @Override
     public <T, E extends Exception> T runTaskWithRetry(TransactionTask<T, E> task) throws E {
+        Preconditions.checkState(!isClosed, "Operations cannot be performed on closed TransactionManager.");
```

Comment:
Might be worth extracting this to a `protected final Boolean checkOpen()` method for subclasses to use.